### PR TITLE
Restrict arrow-like operators to those ending with `->`, `~>` or `=>`

### PR DESCRIPTION
**Pull request by ReneSac**
Updated the news and the manual too. Added further tests in tstrongspaces.nim. 

There were no tests in futures.nim. They were only in tests/macros/tclosuremacro.nim. And they spill `Warning: '(x; y): auto' has no type. Typeless parameters are deprecated; only allowed for 'template' [TypelessParam]` all over the place, but otherwise continues working.

### CODE REVIEW

**Comment by BlaXpirit** [comment]
This pull request is not linked to any discussion.

I don't see how it is beneficial. In fact, I think it is harmful.

**Comment by def-** [comment]
You can search in the irc logs for "arrow":
http://irclogs.nim-lang.org/07-04-2015.html
http://irclogs.nim-lang.org/09-04-2015.html

**Comment by Araq** [comment]
Make it: "Arrow-like if it ends with -> ~> =>" please.

File: compiler/parser.nim [hunk 1]
```diff
@@ -212,7 +212,8 @@ proc getPrecedence(tok: TToken, strongSpaces: bool): int =
     let relevantChar = tok.ident.s[0]
 
     # arrow like?
-    if L > 1 and tok.ident.s[L-1] == '>': return considerStrongSpaces(1)
+    if L > 1 and tok.ident.s[^1] == '>' and tok.ident.s[^2] in {'-', '~', '='}:
```
Comment (Araq) [inline compiler/parser.nim]: Please use 'L-1' and 'L-2' so older compilers can eat it.

File: compiler/parser.nim [hunk 2]
```diff
@@ -212,7 +212,8 @@ proc getPrecedence(tok: TToken, strongSpaces: bool): int =
     let relevantChar = tok.ident.s[0]
 
     # arrow like?
-    if L > 1 and tok.ident.s[L-1] == '>': return considerStrongSpaces(1)
+    if L > 1 and tok.ident.s[L-1] == '>' and
```
Comment (Araq) [inline compiler/parser.nim]: L > 2 instead of L > 1.

Comment (Araq) [inline compiler/parser.nim]: Er, never mind.

**Pull request merged**